be/src/formats/parquet/group_reader.cpp

```diff
@@ -78,11 +78,11 @@ Status GroupReader::get_next(vectorized::ChunkPtr* chunk, size_t* row_count) {
     vectorized::Filter chunk_filter(count, 1);
     DCHECK_EQ(active_chunk->num_rows(), count);
 
-    // dict filter
-    if (has_dict_filter) {
+    // dict filter chunk
+    if (_dict_filter_ctx.has_predicates()) {
         SCOPED_RAW_TIMER(&_param.stats->expr_filter_ns);
         SCOPED_RAW_TIMER(&_param.stats->group_dict_filter_ns);
-        _dict_filter(&active_chunk, &chunk_filter);
+        _dict_filter_ctx.filter_chunk(&active_chunk, &chunk_filter);
```

Comment:
It's more simple if we let `filter_chunk` return bool value to indicate if it has predicates, then we don't need `has_predicates`